File: Sieve_of_Eratosthenes/Python/rrivera1849/sieve.py
def sieve(limit):
    primes = {k:True for k in range(3,limit+1,2)}

    for u in range(3,limit+1,2):
        for i in range(u**2,limit+1,2*u):
            primes[i] = False

    primes[2] = True
    return [u for u in primes if primes[u]==True]


# Testing to see if there are 78498 primes below 1 million

print len(sieve(1000000)) == 78498

File: Sieve_of_Eratosthenes/Lua/Yonaba/sieve.lua

-- Creates an array of consecutive [1..n] integers
local function array(n)
  local t = {}
  for i = 1, n do t[i] = i end
  return t
end

-- Greps truthy values
local function truthy(array)
  local t = {}
  for i,v in ipairs(array) do
    if v then t[#t + 1] = v end
  end
  return t
end

-- Sieve of Eratosthenes
-- returns : an array of primes below a given number n
return function (n)
  local limit = math.floor(math.sqrt(n))
  local primes = array(n)
  for i = 2, limit do
    if primes[i] then
      local j, k = 0, 0
      while j <= n do
        j = i * i + k * i
        k = k + 1
        primes[j] = false
      end
    end
  end
  primes = truthy(primes)
  table.remove(primes, 1)
  table.remove(primes, 1)
  return primes
end
